### PR TITLE
this is release candidate 2.3.3-rc1

### PR DESCRIPTION
**Pull request by lstein**
This includes a number of bug fixes described in the draft release notes.

It also incorporates a modified version of the dialog-based invoke.sh script suggested by JoshuaKimsey:
https://discord.com/channels/1020123559063990373/1089119602425995304

### CODE REVIEW

File: installer/templates/invoke.sh.in [hunk 1]
```diff
@@ -18,78 +21,135 @@ cd "$scriptdir"
...
+	echo "Q - Quit"
+	echo ""
+	read -p "Please enter 1-10, Q: [1] " yn
+	choice=${yn:='1'}
+	do_choice $choice
+    done
+}
+
+if [ "$0" != "bash" ]; then
+    # Dialog seems to be a standard installtion for most Linux distros, but this checks to ensure it is present regardless
```
Comment (JPPhoto) [inline installer/templates/invoke.sh.in:144]: Should be "installation" even though it's a comment.

**Review by JPPhoto — APPROVED** [review]

File: installer/templates/invoke.sh.in (hunk 1 above)
Comment (lstein) [reply installer/templates/invoke.sh.in:144]: Fixed. Thanks.

**Review by ebr — APPROVED** [review]
The interactive `invoke.sh` dialog is really wonderful, though I did have to install the `dialog` utility to make it work (Pop!OS 22.04).

I think at some point we're going to have to rethink these options. The 10-item list is getting overwhelming for the user, and having to maintain it in two places is not ideal. Also it's not necessarily clear that options 5, 6 and 7 have subtle redundancies. But that aside, the textual UX is great.

Approving w/ a grammatical correction

**Review by blessedcoolant — APPROVED** [review]

**Comment by lstein** [comment]
> The interactive `invoke.sh` dialog is really wonderful, though I did have to install the `dialog` utility to make it work (Pop!OS 22.04).
> 
> I think at some point we're going to have to rethink these options. The 10-item list is getting overwhelming for the user, and having to maintain it in two places is not ideal. Also it's not necessarily clear that options 5, 6 and 7 have subtle redundancies. But that aside, the textual UX is great.
> 
> Approving w/ a grammatical correction

In fact this exposes a basic problem with how we do the updates. The updater updates python libraries and the configuration files but not other things inside the invokeai directory such as the launcher scripts. Ideally the launcher script should just be a four-line frontend that changes to the root directory, activates the environment, sets up any environment variables needed, and then calls a python module to do the interactive menu.

@ebr, how's your bandwidth these days? Any interest in refactoring the menu choices?

**Pull request merged**